cmd/erasure-healing.go

```diff
@@ -256,6 +249,21 @@ func (er erasureObjects) healObject(ctx context.Context, bucket string, object s
 		DataBlocks:   len(storageDisks) - er.defaultParityCount,
 	}
 
+	if !opts.NoLock && newObjectLayerFn() != nil {
+		lk := er.NewNSLock(bucket, object)
+		if err := lk.GetLock(ctx, globalOperationTimeout); err != nil {
```

Comment:
@harshavardhana When only a read lock is held several servers will try to heal the same object.

Since there is no coordination if 2 disks are replaced in a set on different servers each will do the heal and they will step on each other. This is easily seen on Windows where tons of file conflicts pop up.

`healObject()` is called in a separate goroutine and will therefore not block calls.